src/components/sync-dialog/SyncDialog.jsx

```diff
@@ -61,7 +66,8 @@ class SyncDialog extends React.Component {
 
     render() {
         const { d2, isOpen } = this.props;
-
+        const { targetInstances } = this.state;
+        const disableSync = targetInstances && _.isEmpty(targetInstances);
         return (
```

Comment:
Minor: let's keep a blank space before the return, it usually keeps function/methods more clear.